repo-scripts/size-analysis/analysis-helper.ts

```diff
@@ -366,3 +387,25 @@ export function writeReportToDirectory(
   }
   fs.writeFileSync(`${directoryPath}/${fileName}`, report);
 }
+
+/**
+ *
+ * @param minimizedDependencies The dependant symbols extracted from a bundle file that doesn't resolve external firebase modules
+ *
+ * @param fullyResolvedDependencies The dependant symbols extracted from a bundle file that resolves external firebase modules
+ *
+ * This function calculates the difference between two dependency sets, the difference is the symbols of external firebase modules
+ * that the current modular export depends on.
+ */
+export function extractExternalDependencies(
```

Comment:
It is going to include symbols we don't care about from within external dependencies. We should only include the top level external dependencies.

For example, for `import { someFunction } from 'externalModule'`, we only want to include `someFunction`, not anything that `someFunction` depends on from within `externalModule`.